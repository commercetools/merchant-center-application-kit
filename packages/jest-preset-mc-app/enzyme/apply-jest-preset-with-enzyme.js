const path = require('path');

const resolveRelativePath = (relativePath) =>
  path.resolve(__dirname, relativePath);

const applyJestPresetWithEnzyme = ({ enzymeAdapterVersion, jestPreset }) => {
  switch (enzymeAdapterVersion) {
    case 16:
    case 17:
      break;
    default:
      throw new Error(
        `Invalid Enzyme adapter version "${enzymeAdapterVersion}". Supported versions are "16", "17".`
      );
  }
  return {
    ...jestPreset,
    // Custom preset config specific for Enzyme.
    snapshotSerializers: [
      ...(jestPreset.snapshotSerializers || []),
      'enzyme-to-json/serializer',
    ],
    setupFilesAfterEnv: [
      ...(jestPreset.setupFilesAfterEnv || []),
      resolveRelativePath(
        `./setup-test-framework-for-enzyme-${enzymeAdapterVersion}.js`
      ),
    ],
  };
};

module.exports = applyJestPresetWithEnzyme;
