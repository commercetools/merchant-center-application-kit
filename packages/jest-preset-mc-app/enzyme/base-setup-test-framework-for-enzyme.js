require('jest-enzyme');

const configureEnzymeExtensions = require('@commercetools/enzyme-extensions');
const commerceToolsEnzymeMatchers = require('@commercetools/jest-enzyme-matchers');
const ShallowWrapper = require('enzyme/ShallowWrapper');

expect.extend({
  toBeComponentWithName(received, actual) {
    const pass = received && received.displayName === actual;
    const message = () =>
      `expected ${received} ${pass ? 'not ' : ''} to be ${actual}`;
    return { message, pass };
  },
});

expect.extend(commerceToolsEnzymeMatchers);
configureEnzymeExtensions(ShallowWrapper);
