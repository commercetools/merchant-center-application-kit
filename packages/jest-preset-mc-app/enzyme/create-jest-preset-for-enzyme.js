const path = require('path');
const defaultPreset = require('../jest-preset');

const resolveRelativePath = (relativePath) =>
  path.resolve(__dirname, relativePath);

const createJestPresetForEnzyme = (version) => ({
  ...defaultPreset,
  snapshotSerializers: ['enzyme-to-json/serializer'],
  setupFilesAfterEnv: [
    resolveRelativePath('../setup-test-framework.js'),
    resolveRelativePath(`./setup-test-framework-for-enzyme-${version}.js`),
  ],
});

module.exports = createJestPresetForEnzyme;
