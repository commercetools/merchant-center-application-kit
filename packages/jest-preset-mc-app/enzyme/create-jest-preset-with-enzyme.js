const path = require('path');

const resolveRelativePath = (relativePath) =>
  path.resolve(__dirname, relativePath);

const createJestPresetForEnzyme = ({ enzymeAdapterVersion, jestPreset }) => {
  switch (enzymeAdapterVersion) {
    case 16:
    case 17:
      break;
    default:
      throw new Error(
        `Invalid Enzyme adapter version "${enzymeAdapterVersion}". Supported versions are "16", "17".`
      );
  }
  return {
    // Use default preset
    ...jestPreset,
    snapshotSerializers: [
      ...(jestPreset.snapshotSerializers || []),
      'enzyme-to-json/serializer',
    ],
    setupFilesAfterEnv: [
      ...(jestPreset.setupFilesAfterEnv || []),
      resolveRelativePath(
        `./setup-test-framework-for-enzyme-${enzymeAdapterVersion}.js`
      ),
    ],
  };
};

module.exports = createJestPresetForEnzyme;
