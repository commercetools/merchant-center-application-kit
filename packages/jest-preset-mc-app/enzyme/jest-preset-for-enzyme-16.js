const createJestPresetForEnzyme = require('./create-jest-preset-for-enzyme');

module.exports = createJestPresetForEnzyme(17);
