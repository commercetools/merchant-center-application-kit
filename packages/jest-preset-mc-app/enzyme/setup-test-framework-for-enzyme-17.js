require('./base-setup-test-framework-for-enzyme');
const Adapter = require('@wojtekmaj/enzyme-adapter-react-17');
const Enzyme = require('enzyme');

Enzyme.configure({ adapter: new Adapter(), disableLifecycleMethods: true });
