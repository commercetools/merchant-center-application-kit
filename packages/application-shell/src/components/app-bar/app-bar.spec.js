import React from 'react';
import { shallow } from 'enzyme';
import { Link } from 'react-router-dom';
import ProjectSwitcher from '../project-switcher';
import UserSettingsMenu from '../user-settings-menu';
import AppBar, { BackToProjectLink } from './app-bar';

jest.mock('../../utils');

const createTestProps = props => ({
  user: {
    projects: {
      total: 0,
    },
    defaultProjectKey: 'test-default-project-key',
    language: 'en',
    firstName: 'John',
    lastName: 'Snow',
    email: 'john.snow@winter.com',
    gravatarHash: '20c9c1b252b46ab49d6f7a4cee9c3e68',
  },
  environment: {
    servedByProxy: false,
  },
  ...props,
});

describe('rendering', () => {
  let props;
  let wrapper;
  beforeEach(() => {
    props = createTestProps();
    wrapper = shallow(<AppBar {...props} />);
  });
  it('should match layout structure', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should render placeholder for "loader-for-requests-in-flight"', () => {
    expect(wrapper).toRender('#loader-for-requests-in-flight');
  });
  it('should render placeholder for "locale-switcher"', () => {
    expect(wrapper).toRender('#locale-switcher');
  });
  describe('<UserSettingsMenu>', () => {
    describe('when user is not defined', () => {
      beforeEach(() => {
        props = createTestProps({ user: null });
        wrapper = shallow(<AppBar {...props} />);
      });
      it('should not render the user settings menu', () => {
        expect(wrapper).not.toRender(UserSettingsMenu);
      });
    });
    describe('when user is loaded', () => {
      beforeEach(() => {
        props = createTestProps();
        wrapper = shallow(<AppBar {...props} />);
      });
      describe('<UserSettingsMenu>', () => {
        it('should render', () => {
          expect(wrapper).toRender(UserSettingsMenu);
        });
        it('should receive "firstName"', () => {
          expect(wrapper.find(UserSettingsMenu)).toHaveProp(
            'firstName',
            'John'
          );
        });
        it('should receive "lastName"', () => {
          expect(wrapper.find(UserSettingsMenu)).toHaveProp('lastName', 'Snow');
        });
        it('should receive "gravatarHash"', () => {
          expect(wrapper.find(UserSettingsMenu)).toHaveProp(
            'gravatarHash',
            '20c9c1b252b46ab49d6f7a4cee9c3e68'
          );
        });
      });
    });
  });
  describe('<ProjectSwitcher>', () => {
    describe('with project key in url', () => {
      describe('when user is not defined', () => {
        beforeEach(() => {
          props = createTestProps({
            user: null,
            projectKeyFromUrl: 'test-project-key',
          });
          wrapper = shallow(<AppBar {...props} />);
        });
        it('should not render <ProjectSwitcher>', () => {
          expect(wrapper).not.toRender(ProjectSwitcher);
        });
      });
      describe('when user is defined', () => {
        describe('when user has no projects', () => {
          beforeEach(() => {
            props = createTestProps({
              user: { ...props.user, projects: { total: 0 } },
              projectKeyFromUrl: 'test-project-key',
            });
            wrapper = shallow(<AppBar {...props} />);
          });
          it('should not render <ProjectSwitcher>', () => {
            expect(wrapper).not.toRender(ProjectSwitcher);
          });
        });
        describe('when user has projects', () => {
          describe('when url contains a `projectKey`', () => {
            describe('<ProjectSwitcher>', () => {
              beforeEach(() => {
                props = createTestProps({
                  user: {
                    ...props.user,
                    projects: { total: 1 },
                  },
                  projectKeyFromUrl: 'test-project-key',
                });
                wrapper = shallow(<AppBar {...props} />);
              });
              it('should render <ProjectSwitcher>', () => {
                expect(wrapper).toRender(ProjectSwitcher);
              });
              it('should pass projectKey to <ProjectSwitcher>', () => {
                expect(wrapper.find(ProjectSwitcher)).toHaveProp(
                  'projectKey',
                  props.projectKeyFromUrl
                );
              });
              it('should pass total to <ProjectSwitcher>', () => {
                expect(wrapper.find(ProjectSwitcher)).toHaveProp('total', 1);
              });
              it('should not render <BackToProjectLink>', () => {
                expect(wrapper).not.toRender(BackToProjectLink);
              });
            });
          });
          describe('when url contains no `projectKey`', () => {
            describe('<ProjectSwitcher>', () => {
              beforeEach(() => {
                props = createTestProps({
                  user: {
                    ...props.user,
                    projects: { total: 1 },
                  },
                  projectKeyFromUrl: null,
                });
                wrapper = shallow(<AppBar {...props} />);
              });
              it('should not render <ProjectSwitcher>', () => {
                expect(wrapper).not.toRender(ProjectSwitcher);
              });
              it('should render <BackToProjectLink>', () => {
                expect(wrapper).toRender(BackToProjectLink);
              });
            });
          });
        });
        describe('when user has not default project', () => {
          describe('<ProjectSwitcher>', () => {
            beforeEach(() => {
              props = createTestProps({
                user: {
                  ...props.user,
                  defaultProjectKey: null,
                },
              });
              wrapper = shallow(<AppBar {...props} />);
            });
            it('should not render <ProjectSwitcher>', () => {
              expect(wrapper).not.toRender(ProjectSwitcher);
            });
            it('should not render <BackToProjectLink>', () => {
              expect(wrapper).not.toRender(BackToProjectLink);
            });
          });
        });
      });
    });
    describe('without project key in url', () => {
      describe('when user is defined', () => {
        describe('when user has projects', () => {
          describe('<ProjectSwitcher>', () => {
            beforeEach(() => {
              props = createTestProps({
                user: {
                  ...props.user,
                  projects: { total: 1 },
                },
              });
              wrapper = shallow(<AppBar {...props} />);
            });
            it('should not render <ProjectSwitcher>', () => {
              expect(wrapper).not.toRender(ProjectSwitcher);
            });

            it('should not render <Link> with path to previous project (default project)', () => {
              expect(wrapper).toRender(Link);
              expect(wrapper.find(Link).last()).toHaveProp(
                'to',
                `/${props.user.defaultProjectKey}`
              );
            });
          });
        });
      });
    });
  });
});
