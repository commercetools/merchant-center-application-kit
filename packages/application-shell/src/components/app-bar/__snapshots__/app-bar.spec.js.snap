// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering should match layout structure 1`] = `
<div
  className="app-bar"
  data-test="top-navigation"
>
  <Inline
    alignItems="flexStart"
    scale="s"
  >
    <div
      className="logo"
    >
      <Link
        replace={false}
        to="/test-default-project-key"
      >
        <img
          alt="Logo"
          className="logo-img"
          src="test-file-stub"
        />
      </Link>
    </div>
    <div
      className="loader-container"
      id="loader-for-requests-in-flight"
    />
  </Inline>
  <div
    className="navigation"
  >
    <Inline
      alignItems="center"
      scale="m"
    >
      <div
        className="switchers"
      >
        <Inline
          alignItems="center"
          scale="s"
        >
          <div
            id="locale-switcher"
          />
          <TextDetail>
            <Link
              replace={false}
              to="/test-default-project-key"
            >
              <Inline
                alignItems="center"
                data-test="back-to-project"
                scale="s"
              >
                <AngleLeftIcon
                  size="small"
                  theme="green"
                />
                <FormattedMessage
                  defaultMessage="Back to project"
                  id="AppBar.backToProjectLink"
                  values={Object {}}
                />
              </Inline>
            </Link>
          </TextDetail>
        </Inline>
      </div>
      <div
        className="spacer"
      />
      <UserSettingsMenu
        email="john.snow@winter.com"
        firstName="John"
        gravatarHash="20c9c1b252b46ab49d6f7a4cee9c3e68"
        lastName="Snow"
        locale="en"
      />
    </Inline>
  </div>
</div>
`;
